src/System.Text.RegularExpressions/tests/PrecompiledRegexScenarioTest.cs

```diff
@@ -0,0 +1,227 @@
+﻿using System;
+using System.Text.RegularExpressions;
+using RegexTestNamespace;
+using Xunit;
+
+namespace System.Text.RegularExpressionsTests
+{
+    public class PrecompiledRegexScenarioTest
+    {
+        [Fact]
+        public void TestPrecompiledRegex()
+        {
+            string text = "asdf134success1245something";
+            RegexTestClass testClass = new RegexTestClass();
+
+            Assert.Equal(1, testClass.Matches(text).Count);
+            Assert.Equal(1, testClass.Match(text).Groups[0].Captures.Count);
+            Assert.Equal(text, testClass.Match(text).Groups[0].Value);
+        }
+    }
+}
+
+namespace RegexTestNamespace
```

Comment:
This was generated with CompileToAssembly API in Desktop. The tools to IL transform the compiled regex from Desktop API is worked upon by @mjsabby. 
@mjsabby had mentioned they will open source the tool.